### PR TITLE
example of factorized jec uncertainty producer

### PR DESCRIPTION
**Pull request by arizzi**

### CODE REVIEW

**Comment by arizzi** [comment]
it is quite slow as far as I can tell (~170 Hz)  likely the double loop can be written better, perhaps adding c++ helpers, plus everything is string based...

**Comment by arizzi** [comment]
@gpetruc running without a cut string seem to crash the nano_postproc.py

File: python/postprocessing/examples/jecUncertProducer.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        pass
+    def analyze(self, event):
+        """process event, return True (go to next module) or False (fail, go to next event)"""
+        jets = Collection(event, "Jet")
+	for u in self.uncerts :
+	    uworker=self.factorizedUncertainties[u]
+	    jetUn=[]
+       	    for j in jets :
+        	uworker.setJetEta(j.eta)
+        	uworker.setJetPt(j.eta)
```
Comment (gpetruc) [inline python/postprocessing/examples/jecUncertProducer.py]: j.pt?

**Comment by arizzi** [comment]
yeah sounds like a good idea !

**Comment by gpetruc** [comment]
https://github.com/gpetruc/nanoAOD-tools/tree/bugfix_pp should fix it

Il 19 Set 2017 20:38, "arizzi" <notifications@github.com> ha scritto:

**Pull request merged**